### PR TITLE
Dependency updates and migration to vite

### LINKED ISSUE #121: Dependency updates
- [x] Get rid of create-react-app dependencies we don’t use
- [x] Use vite. Move from webpack to vite
- [x] Check with Will on github actions changes if any

### PR DESCRIPTION
**Pull request by shane-jung**
Adds Vite and updated npm scripts accordingly. Adds relevant Vite configuration and removes create-react-app artifacts.

Updates .env variables using Vite's naming conventions

Updates outdated packages and resolves dependency vulnerabilities.

Aligns codebase with new package versions-- particularly with authorization with Globus' new SDK version

### CODE REVIEW

**Review by WillEngler — APPROVED** [review]
We'll see for sure when we actually try and deploy it, but this all looks good to me! And I'm really stoked about the formatter being added, not to mention all the other good Vite stuff. Thank you, Shane!

**Review by kjschmidt913 — APPROVED** [review]

File: garden/src/components/ScrollToTop.tsx
Comment (kjschmidt913) [inline garden/src/components/ScrollToTop.tsx:1]: Did you mean to comment this out?

File: garden/src/components/ScrollToTop.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default function ScrollToTop() {
   const { pathname } = useLocation();
 
   useEffect(() => {
-    window.scrollTo(0, 0);
+    // window.scrollTo(0, 0);
```
Comment (kjschmidt913) [inline garden/src/components/ScrollToTop.tsx]: Did you mean to comment this out? Seems like it would make the scroll to the top functionality not work?

Comment (shane-jung) [reply garden/src/components/ScrollToTop.tsx]: You're right! This line was giving me a warning by our test framework, I meant to uncomment it. For now we'll live with the warning

**Pull request merged**